src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution_ArgsToParameters.cs

```diff
@@ -55,6 +55,34 @@ public ImmutableArray<int> ToImmutableArray()
             }
         }
 
+        internal static ImmutableArray<ParameterSymbol> GetParametersIncludingExtensionParameter(Symbol symbol)
+        {
+            Debug.Assert(symbol.GetIsNewExtensionMember());
+            // PROTOTYPE consider optimizing
+            return [symbol.ContainingType.ExtensionParameter, .. symbol.GetParameters()];
+        }
+
+        private static ImmutableArray<RefKind> GetParameterRefKindsIncludingReceiver(Symbol symbol)
+        {
+            Debug.Assert(symbol.GetIsNewExtensionMember());
+            // PROTOTYPE consider optimizing
+            var refKinds = symbol.GetParameterRefKinds();
+            var receiverRefKind = symbol.ContainingType.ExtensionParameter.RefKind;
+            if (refKinds.IsDefault && receiverRefKind == RefKind.None)
+            {
+                return default;
+            }
+
+            return [receiverRefKind, .. refKinds];
```

Comment:
>refKinds [](http://example.com/codeflow?start=40&length=8)

Are we going to get the right result when ```refKinds.IsDefault``` here?

Reply:
Good catch. Thanks
There's an issue at the call-site which was masking this